src/main/java/xyz/nucleoid/plasmid/game/ManagedGameSpace.java

```diff
@@ -22,121 +20,104 @@
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
-import java.util.*;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
 
 /**
- * Represents a unique world with a {@link Game} attached to it.
+ * Represents the space within which a game occurs through attached {@link GameLogic}
  *
  * <p>Each world has a {@link net.minecraft.world.gen.chunk.ChunkGenerator} which is invoked as chunks are requested.
  * It is important to note that not all chunks will be loaded on start, and the game logic must take care to handle this.
- * Players can only be added to this game world through {@link GameWorld#addPlayer} or {@link GameWorld#offerPlayer}.
+ * Players can only be added to this game space through {@link ManagedGameSpace#addPlayer} or {@link ManagedGameSpace#offerPlayer}.
  */
-public final class GameWorld implements AutoCloseable {
-    private static final Map<RegistryKey<World>, GameWorld> DIMENSION_TO_WORLD = new Reference2ObjectOpenHashMap<>();
+public final class ManagedGameSpace implements GameSpace, AutoCloseable {
```

Comment:
Not a fan of the name, but I can't find any better name myself.

Reply:
Yeah- not super happy with it either. Luckily it's only really for internal use, but ideally we would get something better